### ISSUE TITLE
Crash on creating a certificate on Mac M2

### ISSUE THREAD

**Issue by numo68** [chris2511/xca#500]
Trying to create a new certificate signed by an existing homelab certification authority on a Mac M2, Sonoma 14.2.1, xca 2.5.0

```
The following error occurred:
(8pki_x509[]:Censored)
error:0300009C:digital envelope routines::unsupported algorithm
error:068C0100:asn1 encoding routines::malloc failure
error:068C0100:asn1 encoding routines::malloc failure

(pki_x509.cpp:582)
```

**Comment by chris2511** [comment]
Please see #468. Changing all passwords with XCA **2.4.0** solves the issue.

**Comment by numo68** [comment]
Not sure this was the whole problem as I then tried with the Linux version (same version, same hardware, Debian bookworm running in UTM), same file, same CA, same certificate and same private key the Mac version was able to generate.

The only difference was that the database password was changed before ( see #499 ), also with 2.5.0 on Linux.

In any case, if the problem is known, it would be nice to identify it and warn the user.

In any case, thanks for the software that is making the life easier!
